### PR TITLE
Feature Flag remove - USE_NEW_GITHUB_CLIENT_FOR_GET_INSTALLATION

### PR DESCRIPTION
**Pull request by joshkay10**
**Whats in this PR**
Removing feature flag USE_NEW_GITHUB_CLIENT_FOR_GET_INSTALLATION

Removed the octokit object entirely from getInstallations function so had to tidy up calls to getInstallations and tests.

### CODE REVIEW

File: src/routes/jira/configuration/jira-configuration-get.ts [hunk 1]
```diff
@@ -40,18 +38,14 @@ export const getInstallations = async (client: GitHubAPI, subscriptions: Subscri
 	};
 };
 
-const getInstallation = async (client: GitHubAPI, subscription: Subscription, log: Logger): Promise<AppInstallation> => {
+const getInstallation = async (subscription: Subscription, log: Logger): Promise<AppInstallation> => {
 
 	const { jiraHost } = subscription;
-	const useNewGitHubClient = await booleanFlag(BooleanFlags.USE_NEW_GITHUB_CLIENT_FOR_GET_INSTALLATION, false, jiraHost) ;
```
Comment (bgvozdev) [inline src/routes/jira/configuration/jira-configuration-get.ts:46]: Shouldn't the flag also be removed from feature-flags file?

Comment (joshkay10) [reply src/routes/jira/configuration/jira-configuration-get.ts:46]: oo thank you, good catch!

**Review by bgvozdev — APPROVED** [review]

**Review by gxueatlassian — APPROVED** [review]

File: src/routes/jira/configuration/jira-configuration-get.ts [hunk 2]
```diff
@@ -63,12 +57,12 @@ const getInstallation = async (client: GitHubAPI, subscription: Subscription, lo
...
 	} catch (err) {
 		log.error(
-			{ installationId: gitHubInstallationId, error: err, uninstalled: err.code === 404 },
+			{ installationId: gitHubInstallationId, error: err, uninstalled: err.status === 404 },
 			"Failed connection"
 		);
 		statsd.increment(metricError.failedConnection);
 
-		return Promise.reject({ error: err, id: gitHubInstallationId, deleted: err.code === 404 });
+		return Promise.reject({ error: err, id: gitHubInstallationId, deleted: err.status === 404 });
```
Comment (mboudreau) [inline src/routes/jira/configuration/jira-configuration-get.ts:65]: is it from code to status correct?  Fairly certain this is an error of type `AxiosError` and the correct check is `err.response.status`

Comment (joshkay10) [reply src/routes/jira/configuration/jira-configuration-get.ts:65]: investigating

Comment (joshkay10) [reply src/routes/jira/configuration/jira-configuration-get.ts:65]: Good news everyone.
Throws `AxiosError` with `err.response.status` but then gets intercepted and becomes a `GithubClientError` before it reaches the catch block.... at this point it becomes err.status.

We will get failed tests if this changes :)

Comment (mboudreau) [reply src/routes/jira/configuration/jira-configuration-get.ts:65]: ah right.  In that case, maybe change the error type? :)

Comment (joshkay10) [reply src/routes/jira/configuration/jira-configuration-get.ts:65]: It's a try/catch error so typescript wont allow me to type it as anything other than any or undefined.
Unless I misunderstood?

Comment (mboudreau) [reply src/routes/jira/configuration/jira-configuration-get.ts:65]: ah right, stupid try/catch

**Pull request merged**